section2/exercises/if_statements.rb

```diff
@@ -61,5 +66,22 @@
 # Experiment with manipulating the value held within both variables
 # to make sure all above conditions output what you expect.
 
-cups_of_flour = 1
+# The following would be considered a constant:
+flour_required = 2 # in cups
+
+# Error messages
+sauce_error = "Lack of sauce"
+flour_error = "Lack of flour"
+
+cups_of_flour = 3
 has_sauce = true
+
+if cups_of_flour < flour_required && has_sauce == true
+  puts "Error encountered! #{flour_error}"
+  puts "I cannot make a pizza"
+elsif cups_of_flour >= flour_required && has_sauce == false
+  puts "Error encountered! #{sauce_error}"
+  puts "I cannot make a pizza"
+else
+  puts "I can make a pizza!"
+end
```

Comment:
This if/else conditional looks really good. An alternative here (and you'll learn more about this in Turing) is the following:
```
if cups_of_flour < flour_required && has_sauce
  puts "Error encountered! #{flour_error}"
  puts "I cannot make a pizza"
elsif cups_of_flour >= flour_required && !has_sauce
  puts "Error encountered! #{sauce_error}"
  puts "I cannot make a pizza"
else
  puts "I can make a pizza!"
end
```
Since `has_sauce` is a boolean value, you don't need to explicitly pass the `== true` part of the expression since the conditional will just evaluate whether the value of `has_sauce` resolves to a "truthy" value (ie. true, 1, etc.) whereas the `!has_sauce` leverages the `!` or "bang" operator to basically _take the opposite_ of the value of `has_sauce`.